js/components/input.jsx

```diff
@@ -23,23 +23,34 @@ var Input = React.createClass({
   },
 
   getInitialState: function() {
-    return {message: ''};
+    return {
+      message: '',
+      commandCharacter: SettingsStore.commandCharacter()
+    };
+  },
+
+  componentDidMount: function() {
+    SettingsStore.on('change', this.settingsStoreChange);
   },
 
   _onChange: function(event) {
-    this.setState({message: event.target.value});
+    this.setState({message: event.target.value, commandCharacter: this.state.commandCharacter});
   },
 
   _onKeyDown: function(event) {
     if (event.keyCode === KeyCodes.enter) {
       event.preventDefault();
-      if (this.state.message[0] === CommandCharacter) {
+      if (this.state.message[0] === this.state.commandCharacter) {
```

Comment:
hmm.. if it will attempt to re-render but result in no DOM changes, that seems fine, for the sake of being easy to understand (that's what we're talking about, right? re-rendering the virtual DOM is a tiny performance hit, while not re-rendering results in more details for the dev to remember? if so, i will pretty much always prefer easy to understand vs. slight performance benefit).